### PR TITLE
Fixed a collector failure bug and a slow and unnecessary GCS glob

### PR DESCRIPTION
**Pull request by cjllanwarne**

### CODE REVIEW

File: engine/src/main/scala/cromwell/engine/backend/jes/JesEngineFunctions.scala [hunk 1]
```diff
@@ -12,7 +12,7 @@ import scala.util.Try
 class JesWorkflowEngineFunctions(fileSystems: List[FileSystem], context: WorkflowContext) extends WorkflowEngineFunctions(fileSystems, context) {
   override def globPath(glob: String): String = context.root.toAbsolutePath(fileSystems).resolve(JesBackend.globDirectory(glob)).toString
   override def glob(path: String, pattern: String): Seq[String] = {
-    path.toAbsolutePath(fileSystems).asDirectory.glob("**/*") map { _.path.fullPath } filterNot { _.toString == path } toSeq
+    path.toAbsolutePath(fileSystems).asDirectory.list map { _.path.fullPath } toSeq
```
Comment (geoffjentry) [inline engine/src/main/scala/cromwell/engine/backend/jes/JesEngineFunctions.scala]: what kind of seq does that toSeq create?

File: engine/src/main/scala/cromwell/engine/workflow/WorkflowActor.scala [hunk 2]
```diff
@@ -500,11 +499,11 @@ case class WorkflowActor(workflow: WorkflowDescriptor)
...
       val updatedData = data.addPersisting(callKey, status)
       stay() using updatedData
-    case Event(message @ CallFailedRetryable(callKey, events, returnCode, failure), data) if data.isPersistedRunning(callKey) && !data.isPending(callKey) =>
+    case Event(message @ CallFailedRetryable(callKey, events, returnCode, failure), data) if data.isPersistedStatus(callKey, ExecutionStatus.Running) && !data.isPending(callKey) =>
       addCallFailureEvent(callKey.toDatabaseKey, failure.getMessage)
       // Note: Currently Retryable == Preempted. If another reason than preemption arises that requires retrying at this level, this will need to be updated
       handleCallFailedRetryable(callKey, ExecutionStatus.Preempted, returnCode, failure, message, data, events)
-    case Event(message @ CallFailedNonRetryable(callKey, events, returnCode, failure), data) if data.isPersistedRunning(callKey) =>
+    case Event(message @ CallFailedNonRetryable(callKey, events, returnCode, failure), data) if readyToHandleNonRetryableFailure(data, callKey) => // data.isPersistedRunning(callKey) =>
```
Comment (geoffjentry) [inline engine/src/main/scala/cromwell/engine/workflow/WorkflowActor.scala]: Remove commented code, or is that intentional?

**Comment by mcovarr** [comment]
:+1:

File: engine/src/main/scala/cromwell/engine/backend/jes/JesEngineFunctions.scala (hunk 1 above)
Comment (cjllanwarne) [inline engine/src/main/scala/cromwell/engine/backend/jes/JesEngineFunctions.scala]: Discussed in person. No change in this PR

**Comment by geoffjentry** [comment]
👍

**Pull request merged**